subprojects/ear/src/main/java/org/gradle/plugins/ear/EarPlugin.java

```diff
@@ -99,20 +100,25 @@ public FileCollection call() throws Exception {
                         if (project.getPlugins().hasPlugin(JavaPlugin.class)) {
                             return null;
                         } else {
-                            return project.fileTree(earPluginConvention.getAppDirName());
+                            return project.fileTree(task.getAppDirName());
                         }
                     }
                 });
             }
         });
     }
 
-    private void configureWithJavaPluginApplied(final Project project, final EarPluginConvention earPluginConvention, PluginContainer plugins) {
+    private void configureWithJavaPluginApplied(final Project project, final TaskProvider earTask, PluginContainer plugins) {
```

Comment:
`TaskProvider earTask` -> `TaskProvider<Ear> earTask`

Also this `earTask` argument is not used. I guess the intention was to use it below instead of `project.getTasks().named(EarPlugin.EAR_TASK_NAME)`

Reply:
You are right. I've arranged the EarPlugin code and moved the configuration to where the `ear` task is registered.